llama-index-core/llama_index/core/indices/omni_modal/base.py

```diff
@@ -155,6 +158,63 @@ def from_vector_store(
             **kwargs,
         )
 
+    @classmethod
+    def from_documents(
+        cls,
+        documents: Sequence[Document],
+        embed_model: OmniModalEmbeddingBundle[KD, KQ],
+        vector_stores: Optional[Mapping[KD, BasePydanticVectorStore]] = None,
+        # base index params
+        objects: Optional[Sequence[IndexNode]] = None,
+        storage_context: Optional[StorageContext] = None,
+        callback_manager: Optional[CallbackManager] = None,
+        # The default settings use SentenceSplitter which may not be applicable
+        # in multi-modal contexts, so we set the default to no transformations
+        transformations: Optional[Sequence[TransformComponent]] = (),
+        show_progress: bool = False,
+        # vector store index params
+        use_async: bool = False,
+        store_nodes_override: bool = False,
+        insert_batch_size: int = 2048,
+        **kwargs: Any,
+    ) -> "OmniModalVectorStoreIndex[KD, KQ]":
+        storage_context = storage_context or StorageContext.from_defaults()
+        docstore = storage_context.docstore
+
+        if callback_manager is None:
+            callback_manager = callback_manager_from_settings_or_context(Settings, None)
+
+        # Distinguish from the case where an empty sequence is provided.
+        if transformations is None:
```

Comment:
Should we also apply this change to the `BaseIndex`? Imo it's unexpected behaviour that passing `transformation=[]` fails to actually override the default settings.